test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -69,11 +71,29 @@ UcxLog::~UcxLog()
 {
     if (_ss != NULL) {
         (*_ss) << std::endl;
-        std::cout << (*_ss).str();
+        std::cout << _ss->str();
+        check_timeout();
         delete _ss;
     }
 }
 
+void UcxLog::check_timeout() const
+{
+    if (timeout_sec == timeout_inf) {
+        return;
+    }
+
+    double life_time = UcxContext::get_time() - UcxContext::get_time(_tv);
+    if (life_time < timeout_sec) {
+        return;
+    }
+
+    std::cout << "WARNING: too long living (" << life_time
```

Comment:
maybe
`"WARNING: writing the log took too long: " << (log_write_time * 1e6) << " usec"`